lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:js/js.dart';
+import 'package:js/js_util.dart' as js_util;
+
+@JS()
+@staticInterop
+class DomWindow {}
+
+extension DomWindowExtension on DomWindow {
+  external DomDocument get document;
+}
+
+@JS('window')
+external DomWindow get domWindow;
+
+@JS()
+@staticInterop
+class DomDocument {}
+
+extension DomDocumentExtension on DomDocument {
+  external List<DomNode> querySelectorAll(String selectors);
```

Comment:
Hmm, wondering why List<T> seems to work in this case, but not in the other. Going to dig a bit into this...

Reply:
Did we end up getting an answer for this? :)

Reply:
One difference between this instance and the other one is this is lowered by the external extension, where the other one I think you were calling js_util explicitly. I don't know why that would be different, but maybe that's part of the reason?

Reply:
I'm really not sure :(. I thought I saw the issue in the other case, both with the pure external, and with the explicit call to `js_util`, because I thought that was why I tried using `js_util` directly in the first place, but who knows. Maybe it was the explicit call to js_util. Or maybe because in the other case I assigned the return to a temporary, or maybe because in this case the list may have only a single element. Let's sync up offline to see if we want to try and tickle this behavior in a unit test.

In any case, I ended up changing this to `List<Object?>` anyways because that was @yjbanov 's preference. Let's circle back offline to see if we want to investigate this further.